docs/best-practices.md

```diff
@@ -0,0 +1,799 @@
+---
+slug: /best-practices
+---
+
+# Best practices
+
+## Main structure and persistent collections
+
+The main contract structure is marked with `#[near_bindgen]`. It has to be serializable and deserializable with [Borsh](https://borsh.io).
+
+```rust
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct Contract {
+    pub data: String,
+    pub owner_id: AccountId,
+    pub value: u128,
+}
+```
+
+Every time an external method is called, the entire structure has to be deserialized.
```

Comment:
This isn't exactly true, and the correct details are mentioned in the `near_bindgen` section of the docs

Reply:
This is my best understanding, if it's not accurate I'll need some help figuring out what to put: https://github.com/near/sdk-docs/pull/37/commits/ea9fbf46b21966dd5159a1175f6abfe565f504f1

Reply:
The other case I was thinking about is stateless methods (although a rare case)

Reply:
Okay thanks, I might need your help in fleshing that out. Might be something we can add later, if we decide. I'll be addressing the other great comments here to the best of my ability.

Reply:
sounds good, and to be clear what I mean by stateless methods are ones that don't use `self`

Reply:
Interesting, okay. I put an issue here to make sure that doesn't get lost. https://github.com/near/sdk-docs/issues/39